### ISSUE TITLE
Background training infrastructure: GitHub Actions + Mac mini

### ISSUE THREAD

**Issue by everettVT** [VangelisTech/archetype#48]
## Problem

The synth engine recursion loop converges in ~2min on 35 segments locally. But scaling to larger corpora (100s-1000s of segments across conversation history, codebase diffs, ECS snapshots) needs:

1. **Longer training runs** — more epochs, larger batches, spectral clustering (MLX power iteration) instead of k-means
2. **Parallel worlds** — explore different hyperparameters (n_clusters, learning rate, diffusion steps, label sources) simultaneously
3. **Continuous improvement** — schedule training on new data as conversations accumulate

### Option A: GitHub Actions (CPU-only, free tier)

- Workflow triggered on push to `examples/synth/` or manually via `workflow_dispatch`
- Matrix strategy: spin up parallel jobs for different configs (`{clusters: [3,4,6,8], method: [kmeans, spectral]}`)
- Upload model artifacts + `synth_results.json` as workflow artifacts
- Compare results across runs via a summary step

### Option B: Mac mini via Tailscale (MLX GPU)

- Self-hosted runner on the Mac mini (already on Tailscale network)
- MLX spectral clustering runs on Apple GPU — the power iteration benchmark showed 44x speedup over sklearn
- Long-running training: `recurse.py --cycles 20 --epochs 50` with spectral processor variant
- Results pushed to a branch or uploaded as artifacts

### Option C: Hybrid

- Actions for the CPU-bound work (triplet generation, sklearn baseline, regression tests)
- Mac mini for MLX GPU work (spectral clustering, large-scale embedding)
- Coordinate via a shared artifact store or LanceDB branch

## Archetype parallel worlds pattern

The ECS architecture supports this natively. Fork a world, run different processors, compare:

```python
# Spawn parallel worlds with different clustering strategies
worlds = {
    "kmeans_k4": (ClusterProcessor(n_clusters=4), {}),
    "kmeans_k8": (ClusterProcessor(n_clusters=8), {}),
    "spectral_k4": (SpectralClusterProcessor(n_clusters=4), {}),
}

for name, (proc, results) in worlds.items():
    world = create_world(f"synth_{name}")
    world.add_processor(proc)
    spawn_entities(world, segments)
    world.run(RunConfig(num_steps=5))
    results[name] = world.query(...)
```

Each world is independent — perfect for parallelization on Actions or across machines.

## Next steps

- [ ] Set up self-hosted runner on Mac mini
- [ ] Create `.github/workflows/synth-train.yml` with matrix strategy
- [ ] Wire `SpectralClusterProcessor` (MLX power iteration) into `recurse.py` as methodology 4
- [ ] Build comparison dashboard from cross-world results

/cc @claude
